File: algorithms/204.swift
// https://leetcode.com/problems/count-primes

class Solution {
    func countPrimes(n: Int) -> Int {
        guard n > 2 else { return 0 }
        var isPrime = [Bool](count: n, repeatedValue: true)
        isPrime[0..<2] = [false, false]
        for i in 0..<Int(sqrt(Double(n)))+1 {
            if isPrime[i] {
                (i*i).stride(to: n, by: i).map { isPrime[$0] = false }
            }
        }
        return isPrime.filter { $0 == true }.count
    }
}

File: algorithms/204.py
# https://leetcode.com/problems/count-primes


class Solution(object):
    def countPrimes(self, n):
        """
        :type n: int
        :rtype: int
        """
        if n <= 2:
            return 0

        is_prime = [True] * n
        is_prime[:2] = [False, False]
        for i in range(int((n - 1)**0.5) + 1):
            if is_prime[i]:
                is_prime[i**2::i] = [False] * len(is_prime[i**2::i])
        return is_prime.count(True)
